### PR TITLE
(maint) Add baseline documentation for i18n support

### PR DESCRIPTION
**Pull request by ahenroid**

### CODE REVIEW

**Comment by ahenroid** [comment]
@kylog following up on your request to document i18n best practices for C++, I realized that the Leatherman docs actually cover this fairly well already. So, based on a conversation with @MikaelSmith, I'm making minor edits here and including a pointer to the Leatherman docs.

File: 01-Getting-Started.md [hunk 1]
```diff
@@ -299,6 +301,14 @@ Assume POSIX functions aren't available on Windows, and prefer Boost or C++ stan
 
 TODO: Add ways to get common paths on Windows.
 
+### Internationalization (i18n)
+
+In light of today's global marketplace for software, internationalization support (abbreviated "i18n") is critical, and Puppet is making strides towards full i18n-compliance across all products. At a minimum, compliant projects must address translation of user-visible strings and provide correct handling of Unicode (UTF-8).
+
+#### Extracting and Translating Text
+
+Leatherman provides support for marking translatable strings and generating and using message catalogs (.pot files) based on the `gettext` library. Please refer to the [`Leatherman documentation`](https://github.com/puppetlabs/leatherman#internationalization-i18n) for details.
```
Comment (MikaelSmith) [inline 01-Getting-Started.md]: I'd change translatable to localizable (as it may include translation, but also changing formatting for specific locales).

Comment (ahenroid) [inline 01-Getting-Started.md]: Ah, good point. Done.

**Comment by MikaelSmith** [comment]
Please add a link to the Internationalization section to the main README as well.

**Comment by ahenroid** [comment]
Ah! Sorry I missed the README update. Just fixed. Normally, I believe, doctoc would have
made this update, but I am manually updating TOCs because the version of doctoc I have
here was stripping out links and otherwise breaking *.md.

**Comment by MikaelSmith** [comment]
There are some weird things with doctoc and this anyway, I think I usually have to manually futz with it.

**Pull request merged**